### PR TITLE
Refactor Phalcon\Db\Adapter\Pdo::query

### PR DESCRIPTION
**Pull request by cq-z**
Hello!

* Type: new feature
* Link to issue:

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:

- Refactored `Phalcon\Db\Adapter\Pdo::query` to use PDO's prepare and execute.
- `Phalcon\Db\Adapter::fetchAll` to use PDO's fetchAll

Thanks

### CODE REVIEW

File: CHANGELOG-3.4.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 - Fixed `Phalcon\Tag` so it unsets `parameters` before passing options array to `self::renderAttributes`
 - Fixed  `\Phalcon\Http\Response::setFileToSend` filename; when file downloaded it had an extra `_`
 - Added the ability to explicitly define nullable columns (especially timestamp ones). [#13099](https://github.com/phalcon/cphalcon/issues/13099)
+- Refactor `Phalcon\Db\Adapter\Pdo::query` in fetchOne get All fetch One, `Phalcon\Db\Adapter::queryAll` use `pdo::queryAll`
```
Comment (niden) [inline CHANGELOG-3.4.md]: Refactored `Phalcon\Db\Adapter\Pdo::query` to use PDO's `prepare` and `execute`. `Phalcon\Db\Adapter::fetchAll` to use PDO's `fetchAll`

**Review by niden — CHANGES_REQUESTED** [review]

File: phalcon/db/adapter/pdo.zep [hunk 2]
```diff
@@ -360,7 +360,8 @@ abstract class Pdo extends Adapter
 				let statement = this->executePrepared(statement, bindParams, bindTypes);
 			}
 		} else {
-			let statement = pdo->query(sqlStatement);
+			let statement = pdo->prepare(sqlStatement);
+			statement->execute();
 		}
```
Comment (niden) [inline phalcon/db/adapter/pdo.zep]: We can reuse existing code here and do some minor refactoring to use `prepare` for both cases.

The code can be refactored as follows:
```php
	public function query(string! sqlStatement, var bindParams = null, var bindTypes = null) -> <ResultInterface> | boolean
	{
		var eventsManager, pdo, statement, params, types;

		let eventsManager = <ManagerInterface> this->_eventsManager;

		/**
		 * Execute the beforeQuery event if an EventsManager is available
		 */
		if typeof eventsManager == "object" {
			let this->_sqlStatement = sqlStatement,
				this->_sqlVariables = bindParams,
				this->_sqlBindTypes = bindTypes;
			if eventsManager->fire("db:beforeQuery", this) === false {
				return false;
			}
		}

		let pdo = <\Pdo> this->_pdo;
		if typeof bindParams == "array" {
			let params = bindParams;
			let types = bindTypes;
		} else {
			let params = [];
			let types = [];
		}
			
		let statement = pdo->prepare(sqlStatement);
		if typeof statement == "object" {
			let statement = this->executePrepared(statement, params, types);
		}

		/**
		 * Execute the afterQuery event if an EventsManager is available
		 */
		if typeof statement == "object" {
			if typeof eventsManager == "object" {
				eventsManager->fire("db:afterQuery", this);
			}
			return new ResultPdo(this, statement, sqlStatement, bindParams, bindTypes);
		}

		return statement;
	}
```

This way we can utilize the `executePrepared` and `prepare` methods of the class.

**Review by niden — CHANGES_REQUESTED** [review]

File: phalcon/db/adapter/pdo.zep (hunk 2 above)
Comment (cq-z) [reply phalcon/db/adapter/pdo.zep]: good!

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
Thank you !!

**Pull request merged**